### DISCUSSION TITLE
How to tell which file slows ripgrep down?

### DISCUSSION THREAD

**Discussion by openaphid** [BurntSushi/ripgrep#2022] [Q&A] [answered]
When ripgrep gets stuck, is there a verbose mode to trace which file was the root cause?

**Comment by BurntSushi** [comment, marked as answer]
You can try the `--trace` flag.

Another approach is a binary search of sorts. Keep removing things from your search until it gets faster. When it gets faster, check that what you removed makes ripgrep slow. Then repeat.

**Reply by openaphid** [reply]
Sorry for my late reply. 

I tried  `--trace` and `--debug` flags. The additional logs seem not relevant when `rg` gets stuck. 

```
TRACE|grep_searcher::searcher|crates/searcher/src/searcher/mod.rs:680: Some("PREVIOUS_FILE, not the ongoing file"): searching using generic reader
TRACE|grep_searcher::searcher|crates/searcher/src/searcher/mod.rs:729: generic reader: searching via roll buffer strategy
TRACE|grep_searcher::searcher::core|crates/searcher/src/searcher/core.rs:56: searcher core: will use fast line searcher
```

It's a large code repo which makes it difficult to do a binary search. I was hoping there is an easy way to spot slow files to add to exclude list.

Thanks for the help. I'll try to understand the code of `rg` and add logs for this specific case.
